### ISSUE TITLE
daal4py patches to scikit-learn cause test_dtype_match to fail

### ISSUE THREAD

**Issue by oleksandr-pavlyk** [IntelPython/daal4py#86]
With https://github.com/IntelPython/daal4py/pull/84 daal4py is used to compute logistic loss and its gradient with solver='newton-cg'. This causes 

```
python -m daal4py -m pytest --pyargs sklearn.linear_model.tests.test_logistic::test_dtype_match -ra
```

to fail, so it is currently being deselected, see [deselected_tests.yaml#L31](https://github.com/IntelPython/daal4py/blob/master/deselected_tests.yaml#L31).

This issue is to document the deselection and the failure:

```text
>       assert_allclose(lr_32.coef_, lr_64.coef_.astype(np.float32), rtol=rtol)
E       AssertionError:
E       Not equal to tolerance rtol=1e-06, atol=0
E
E       Mismatch: 100%
E       Max absolute difference: 0.00011081
E       Max relative difference: 0.00011659
E        x: array([[0.950293, 0.723663]], dtype=float32)
E        y: array([[0.950404, 0.723695]], dtype=float32)
```

**Comment by david-cortes-intel** [comment]
This is expected behavior, since there are differences in how both libraries handle large/small probabilities that could lead to these differences. The solve `newton-cg` is also only patched under preview mode node.
